### DISCUSSION TITLE
The whole repository inside each component on server (self-hosted Weblate)

### DISCUSSION THREAD

**Discussion by alinakidenko** [WeblateOrg/weblate#12270] [Q&A]
Hello,

I have a question regarding the use of translation components with internal URLs.

When I create translation components and link them through an internal URL to a single component, it doesn’t reduce server disk space. **Each component contains the entire repository**, making them quite large. For instance, the main component that other components link to is 179 MB, and each individual component is 82 MB. I have only three translation components so far: one has 16 one-line strings, the second has 64, and the third has 116.

We are using a self-hosted Weblate instance. Weblate project is connected to a GitHub repository. The linked components are marked with an icon indicating their linked status (see screenshot). Additionally, the "Manage/Status of repositories" section on the Weblate page shows information confirming that the components are linked. (see the second screenshot)

![image](https://github.com/user-attachments/assets/193ab48c-c3d3-4b89-ba6d-21dbc2386e94)

![image](https://github.com/user-attachments/assets/b223fad4-1bff-44a6-b6d0-b486bf64c12f)

Could anyone provide insights into whether this behavior is expected, or if there might be a way to optimize disk space usage? Any advice or suggestions would be greatly appreciated.

Thank you in advance for your help :)

**Comment by nijel** [comment]
Did you change the setup to use linked repositories? In that case the cleanup might take some time.

**Reply by alinakidenko** [reply]
No, I create the components via REST API with internal links from the beginning. 

All settings in Weblate after creating components via the REST API seem correct, but on the server, it takes up a lot of space for every component.

For example, this is a setting for a linked component called "condition":

![image](https://github.com/user-attachments/assets/e16a9dc4-b332-4e32-8986-c58e052f9b5f)

This component seems to be correctly linked to another component, but on the server I have this:

![image](https://github.com/user-attachments/assets/18d3b2c3-3de3-41b7-834a-5485a47c3f76)

![image](https://github.com/user-attachments/assets/a438e183-9fd4-4732-b950-c1079e52f97b)


Here is the body for request:

```
const body = {
    name: fileNameWoExt,
    slug: createSlug(fileNameWoExt),
    category: categoriesByPath.get(categoryPath).url,

    vcs: "git",
    repo,
    branch,
    ...(fileNameWoExt !== "dashboard" && {
      linked_component: `weblate://${syncEnv.weblateProjectName}/${branch}/abra-framework/dashboard`
    }),

    file_format: "json-nested",
    source_language: lang,
    language_code_style: "bcp_long",
    filemask,
    template: base,
    new_lang: "add",

    allow_translation_propagation: "false",
    commit_pending_age: "0",
    manage_units: "false"
  };
```

**Reply by alinakidenko** [reply]
When components are created via the REST API with internal links, the server performs a cleanup roughly two days later at 2:40 AM. Is it possible to create components with internal links from the start via the REST API, so we can save space and reduce waiting time?

P.S. We are creating the main component first so that other components can be linked to it.

**Reply by nijel** [reply]
It should not clone the repository, and it behaves correctly in the test suite where I've just added it in e8e1a56090.
